chef-config/lib/chef-config/config.rb

```diff
@@ -513,7 +513,32 @@ def self.path_accessible?(path)
     default :recipe_url, nil
 
     # Set to true if Chef is to set OpenSSL to run in FIPS mode
-    default(:fips) { ENV["CHEF_FIPS"] == "1" }
+    default(:fips) do
+      !ENV["CHEF_FIPS"].nil? || check_fips_via_ohai
+    end
+
+    # we want to synchronize this ohai call because ohai is not thread safe
+    # if this gets called in a mulithreaded context, each thread's ohai instance
+    # will call reset_system while other threads are loading plugins
+    # the destructive power of reset_system is scoped to the module and not to the instance
+    def self.check_fips_via_ohai
+      @@sync ||= Mutex.new
```

Comment:
`||=` in ruby is two ruby instructions which can get context switched, so it's best to avoid it when using threads. I'd recommend instead putting `@@sync = Mutex.new` in the class body.

Also, do you need the extra behaviors of class variables here (instead of a class instance variable)?